stellargraph/core/element_data.py

```diff
@@ -331,28 +364,53 @@ def __init__(self, shared, type_starts):
 
     def _init_directed_adj_lists(self):
         # record the edge ilocs of incoming, outgoing and both-direction edges
-        in_dict = {}
-        out_dict = {}
+        if len(self.targets) == 0:
```

Comment:
Why is this special case required?

Reply:
anningly `.max` breaks on an empty array 😡